### PR TITLE
Updated DMTCP for truncate feature.

### PR DESCRIPTION
**Pull request by karya0**

### CODE REVIEW

**Comment by karya0** [comment]
@dahongli @JainTwinkle : Please update the file paths in mpi_plugin.cpp so we can merge this PR.

**Comment by dahongli** [comment]
> @dahongli @JainTwinkle : Please update the file paths in mpi_plugin.cpp so we can merge this PR.

@karya0 Please use the following update to check the file path.
`diff --git a/mpi-proxy-split/mpi_plugin.cpp b/mpi-proxy-split/mpi_plugin.cpp
index bd9b75d..4fd1a91 100644
--- a/mpi-proxy-split/mpi_plugin.cpp
+++ b/mpi-proxy-split/mpi_plugin.cpp
@@ -126,10 +126,20 @@ dmtcp_skip_memory_region_ckpting(const ProcMapsArea *area)
   return 0;
 }
 
+#define P2P_LOG_MSG "p2p_log_%d.txt"
+#define P2P_LOG_REQUEST "p2p_log_request_%d.txt"
 EXTERNC int
 dmtcp_skip_truncate_file_at_restart(const char* path)
 {
-  if (strstr(path, "/some/path/to/log")) {
+  char p2p_log_name[100];
+  char p2p_log_request_name[100];
+  int rank;
+
+  MPI_Comm_rank(MPI_COMM_WORLD, &rank);
+  snprintf(p2p_log_name, sizeof(p2p_log_name)-1, P2P_LOG_MSG, rank);
+  snprintf(p2p_log_request_name, sizeof(p2p_log_request_name)-1, P2P_LOG_REQUEST, rank);
+  if (strstr(path, p2p_log_name) == NULL &&
+      strstr(path, p2p_log_request_name) == NULL) {
     // Do not truncate this file.
     return 1;
   }
`

**Comment by karya0** [comment]
@dahongli: Thanks for the paths. I have updated the PR to reflect the changes. Please take a look.

**Comment by dahongli** [comment]
@karya0 I should check the log filename the other way around. Please update the change. Thanks.
```diff --git a/mpi-proxy-split/mpi_plugin.cpp b/mpi-proxy-split/mpi_plugin.cpp
index 051a6e9f8..40a37b9e9 100644
--- a/mpi-proxy-split/mpi_plugin.cpp
+++ b/mpi-proxy-split/mpi_plugin.cpp
@@ -139,8 +139,8 @@ dmtcp_skip_memory_region_ckpting(const ProcMapsArea *area)
   snprintf(p2p_log_name, sizeof(p2p_log_name) - 1, P2P_LOG_MSG, rank);
   snprintf(p2p_log_request_name, sizeof(p2p_log_request_name)-1, P2P_LOG_REQUEST, rank);
 
-  if (strstr(path, p2p_log_name) == NULL &&
-      strstr(path, p2p_log_request_name) == NULL) {
+  if (strstr(path, p2p_log_name) ||
+      strstr(path, p2p_log_request_name)) {
     // Do not truncate this file.
     return 1;
   }
```

**Comment by karya0** [comment]
> ```diff
> memory_region_ckpting(const ProcMapsArea *area)
>    snprintf(p2p_log_name, sizeof(p2p_log_name) - 1, P2P_LOG_MSG, rank);
>    snprintf(p2p_log_request_name, sizeof(p2p_log_request_name)-1, P2P_LOG_REQUEST, rank);
>  
> -  if (strstr(path, p2p_log_name) == NULL &&
> -      strstr(path, p2p_log_request_name) == NULL) {
> +  if (strstr(path, p2p_log_name
> ```

Done

File: mpi-proxy-split/mpi_plugin.cpp [hunk 1]
```diff
@@ -126,6 +126,29 @@ dmtcp_skip_memory_region_ckpting(const ProcMapsArea *area)
...
+  constexpr const char* P2P_LOG_REQUEST = "p2p_log_request_%d.txt";
+  char p2p_log_name[100];
+  char p2p_log_request_name[100];
+  int rank;
+
+  MPI_Comm_rank(MPI_COMM_WORLD, &rank);
+  snprintf(p2p_log_name, sizeof(p2p_log_name) - 1, P2P_LOG_MSG, rank);
+  snprintf(p2p_log_request_name, sizeof(p2p_log_request_name)-1, P2P_LOG_REQUEST, rank);
+
+  if (strstr(path, p2p_log_name) &&
```
Comment (dahongli) [inline mpi-proxy-split/mpi_plugin.cpp:142]: Replace "&&" with "||" here.

Comment (karya0) [reply mpi-proxy-split/mpi_plugin.cpp]: Done

**Review by dahongli — APPROVED** [review]
Looks good. Thanks Kapil!

**Comment by karya0** [comment]
@dahongli: Is it ok to merge this PR?

**Comment by JainTwinkle** [comment]
@karya0 I don't have access to Jira to check what exactly is failing. I tried building mana locally after applying this PR. I see the following error:
```

**Comment by karya0** [comment]
Thanks for checking it out, @JainTwinkle ! Apparently, DMTCP uses NEXT_FNC for finding the symbol in the next library, whereas MANA uses NEXT_FUNC. I accidentally used NEXT_FUNC instead of NEXT_FNC -- we need better names for both.

**Pull request merged**